presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -310,17 +310,18 @@ public boolean isNull(int position)
     }
 
     @Override
-    public Block getPositions(int[] positions)
+    public Block getPositions(int[] positions, int offset, int length)
     {
-        int length = positions.length;
+        checkValidPositions(positions, offset, length, positionCount);
```

Comment:
Checking `offset` and `length` via  `checkValidPositionsArray` is enough. `checkValidPositions` also checks contents of the array, but this is redundant here.

Reply:
See also #9598 > `Remove BlockUtil.checkValidPositions`